atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -81,10 +90,59 @@
     private final SqlConnectionSupplier connections;
 
     public static DbKvs create(DbKeyValueServiceConfig config) {
-        return new DbKvs(config, config.tableFactorySupplier().get(), config.sqlConnectionSupplier());
+        Preconditions.checkArgument(config.connection().isPresent(),
```

Comment:
Any particular reason we do this config check here as compared to during the config creation?  Ie add a @Check to DbKeyValueServiceConfig (or it's impls, though it appears connection would be valuable for both).